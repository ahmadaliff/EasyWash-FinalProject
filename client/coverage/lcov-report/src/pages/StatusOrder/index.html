
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/pages/StatusOrder</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../../../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../../../index.html">All files</a> src/pages/StatusOrder</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.16% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>50/106</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">36.11% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>13/36</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">38.23% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>13/34</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">47.91% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>46/96</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input oninput="onInput()" type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line low'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="actions.js"><a href="actions.js.html">actions.js</a></td>
	<td data-value="75" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 75%"></div><div class="cover-empty" style="width: 25%"></div></div>
	</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="16" class="abs medium">12/16</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="8" class="abs medium">4/8</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="8" class="abs high">8/8</td>
	</tr>

<tr>
	<td class="file high" data-value="constants.js"><a href="constants.js.html">constants.js</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.jsx"><a href="index.jsx.html">index.jsx</a></td>
	<td data-value="79.16" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 79%"></div><div class="cover-empty" style="width: 21%"></div></div>
	</td>
	<td data-value="79.16" class="pct medium">79.16%</td>
	<td data-value="24" class="abs medium">19/24</td>
	<td data-value="70.58" class="pct medium">70.58%</td>
	<td data-value="17" class="abs medium">12/17</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="82.6" class="pct high">82.6%</td>
	<td data-value="23" class="abs high">19/23</td>
	</tr>

<tr>
	<td class="file low" data-value="reducer.js"><a href="reducer.js.html">reducer.js</a></td>
	<td data-value="30.76" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 30%"></div><div class="cover-empty" style="width: 70%"></div></div>
	</td>
	<td data-value="30.76" class="pct low">30.76%</td>
	<td data-value="13" class="abs low">4/13</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="5" class="abs low">1/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="30.76" class="pct low">30.76%</td>
	<td data-value="13" class="abs low">4/13</td>
	</tr>

<tr>
	<td class="file low" data-value="saga.js"><a href="saga.js.html">saga.js</a></td>
	<td data-value="13.63" class="pic low">
	<div class="chart"><div class="cover-fill" style="width: 13%"></div><div class="cover-empty" style="width: 87%"></div></div>
	</td>
	<td data-value="13.63" class="pct low">13.63%</td>
	<td data-value="44" class="abs low">6/44</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="7.14" class="pct low">7.14%</td>
	<td data-value="14" class="abs low">1/14</td>
	<td data-value="13.95" class="pct low">13.95%</td>
	<td data-value="43" class="abs low">6/43</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-01-03T03:15:37.528Z
            </div>
        <script src="../../../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../../../sorter.js"></script>
        <script src="../../../block-navigation.js"></script>
    </body>
</html>
    