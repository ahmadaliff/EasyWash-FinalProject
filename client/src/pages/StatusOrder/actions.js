import {
  CONNECTED_SOCKET,
  CONNECT_SOCKET,
  DISCONNECT_SOCKET,
  GET_STATUS,
  PAY,
  SET_STATUS,
  STATUS_UPDATED,
  WATCH_STATUS_UPDATES,
} from './constants';

export const actionGetStatus = (orderId) => ({
  type: GET_STATUS,
  orderId,
});

export const actionSetStatus = (data) => ({
  type: SET_STATUS,
  data,
});

export const actionStatusUpdated = (data) => ({
  type: STATUS_UPDATED,
  data,
});

export const actionConnectSocket = () => ({
  type: CONNECT_SOCKET,
});

export const actionConnectedSocket = (data) => ({
  type: CONNECTED_SOCKET,
  data,
});

export const actionDisconnectSocket = () => ({
  type: DISCONNECT_SOCKET,
});

export const watchStatusUpdates = (orderId) => ({
  type: WATCH_STATUS_UPDATES,
  orderId,
});

export const actionPay = (orderId) => ({
  type: PAY,
  orderId,
});
