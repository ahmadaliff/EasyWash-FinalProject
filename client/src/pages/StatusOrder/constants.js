export const GET_STATUS = 'status/GET_STATUS';
export const SET_STATUS = 'status/SET_STATUS';
export const STATUS_UPDATED = 'status/STATUS_UPDATED';
export const RESET_STATUS = 'status/RESET_STATUS';
export const CONNECT_SOCKET = 'status/CONNECT_SOCKET';
export const CONNECTED_SOCKET = 'status/CONNECTED_SOCKET';
export const DISCONNECT_SOCKET = 'status/DISCONNECT_SOCKET';
export const WATCH_STATUS_UPDATES = 'status/WATCH_STATUS_UPDATES';
export const PAY = 'status/PAY';
