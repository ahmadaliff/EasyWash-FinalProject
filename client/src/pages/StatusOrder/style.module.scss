@import '@styles/base';

.mainWrap {
  @include directionY;
  padding: 1rem;
  @include apply-at-md-size {
    padding: 2rem 5rem 2rem 5rem;
  }
  min-height: var(--min-height);
  align-items: center;
  box-sizing: border-box;
  position: relative;
  @include apply-at-md-size {
    min-height: var(--min-height-sm);
  }
  * {
    box-sizing: border-box;
  }
  .header {
    @include directionX;
    justify-content: space-between;
    width: 100%;
    align-items: center;
    .backButton {
      margin-left: auto;
      background-color: var(--color-primary);
      border-radius: 10rem;
      padding: 0.25rem 1rem 0.25rem 1rem;
      text-transform: capitalize;
      font-size: medium;
      &:hover {
        background-color: var(--color-primary);
      }
    }
  }
  .card {
    width: 100%;
    position: relative;
    .field {
      @include directionX;
      align-items: center;
      font-size: 0.75rem;
      font-weight: 500;
      .label {
        float: left;
        font-weight: 700;
        width: 10rem;
      }
      .text {
        width: 100%;
      }
    }
    .service {
      @include directionX;
      align-items: center;
      gap: 0.125rem;
      padding: 0.25rem;
      width: 100%;
      .serviceInfoWrap {
        width: 100%;
        p {
          margin: 0;
          font-size: small;
        }
        .quantity {
          @include directionX;
          align-items: center;
          justify-content: space-between;
          font-size: small;
        }
        .serviceInfo {
          @include directionX;
          justify-content: space-between;
          .name {
            @include directionY;
            .type {
              font-size: x-small;
            }
          }
        }
      }
    }
    .downloadButton {
      background-color: var(--color-primary);
      padding: 0.25rem 1rem 0.25rem 1rem;
      text-transform: capitalize;
      font-size: small;
      &:hover {
        background-color: var(--color-primary);
      }
    }
    .cancelButton,
    .payButton {
      margin-top: 1rem;
      width: 100%;
      background-color: rgba($color: red, $alpha: 0.7);
      border-radius: 10rem;
      padding: 0.25rem 1rem 0.25rem 1rem;
      text-transform: capitalize;
      font-size: small;
    }
    .payButton {
      background-color: var(--color-primary);
    }
  }
}
