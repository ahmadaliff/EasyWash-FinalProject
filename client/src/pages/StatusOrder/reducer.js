import { produce } from 'immer';
import { SET_STATUS, STATUS_UPDATED } from './constants';

export const initialState = {
  order: null,
  status: null,
};

const statusOrderReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_STATUS:
        draft.order = action.data;
        break;
      case STATUS_UPDATED: {
        const order = { ...state.order };
        order.status = action.data;
        draft.status = order;
        break;
      }
    }
  });

export default statusOrderReducer;
