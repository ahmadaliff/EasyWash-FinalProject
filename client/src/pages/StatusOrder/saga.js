import { call, put, takeEvery } from 'redux-saga/effects';
import { eventChannel } from 'redux-saga';
import toast from 'react-hot-toast';
import io from 'socket.io-client';

import intlHelper from '@utils/intlHelper';

import { CONNECT_SOCKET, DISCONNECT_SOCKET, GET_STATUS, WATCH_STATUS_UPDATES } from '@pages/StatusOrder/constants';
import { showPopup } from '@containers/App/actions';
import { actionSetStatus, actionStatusUpdated } from '@pages/StatusOrder/actions';
import { apiGetOrderById } from '@domain/api';
import config from '@config/index';

const createSocketChannel = (socket) =>
  eventChannel((emit) => {
    const handler = (data) => emit(data);
    socket.on('statusUpdated', handler);

    return () => {
      socket.off('statusUpdated', handler);
    };
  });

function* connectSocket({ orderId }) {
  const socket = yield call(io, `${config.api.server}/user/${orderId}`);
  yield put({ type: 'SOCKET_CONNECTED', socket });
}

function* disconnectSocket({ orderId }) {
  const socket = yield call(io, `${config.api.server}/user/${orderId}`);
  if (socket) {
    socket.disconnect();
  }
}

function* sagaGetStatus({ orderId }) {
  try {
    const response = yield call(apiGetOrderById, orderId);
    yield put(actionSetStatus(response.data));
  } catch (error) {
    if (error?.response?.status === 400 || error?.response?.status === 404) {
      toast.error(intlHelper({ message: error.response.data.message }));
    } else {
      yield put(showPopup());
    }
  }
}

function* handleStatusUpdate(payload) {
  yield put(actionStatusUpdated(payload));
}

function* watchStatusUpdates({ orderId }) {
  const socket = yield call(io, `${config.api.server}/user/${orderId}`);
  const channel = yield call(createSocketChannel, socket);
  yield takeEvery(channel, handleStatusUpdate);
}

export function* statusOrderSaga() {
  yield takeEvery(GET_STATUS, sagaGetStatus);
  yield takeEvery(CONNECT_SOCKET, connectSocket);
  yield takeEvery(DISCONNECT_SOCKET, disconnectSocket);
  yield takeEvery(WATCH_STATUS_UPDATES, watchStatusUpdates);
}
