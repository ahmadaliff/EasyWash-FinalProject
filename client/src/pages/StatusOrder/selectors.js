import { createSelector } from 'reselect';
import { initialState } from '@pages/StatusOrder/reducer';

const selectStatusOrderState = (state) => state.statusOrder || initialState;

export const selectOrder = createSelector(selectStatusOrderState, (state) => state.order);
