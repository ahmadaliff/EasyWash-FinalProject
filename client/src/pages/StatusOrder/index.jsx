import { PropTypes } from 'prop-types';
import { useEffect } from 'react';
import { useParams } from 'react-router-dom';
import { connect, useDispatch } from 'react-redux';

import {
  actionConnectSocket,
  actionDisconnectSocket,
  actionGetStatus,
  watchStatusUpdates,
} from '@pages/StatusOrder/actions';
import { createStructuredSelector } from 'reselect';
import { selectOrder } from './selectors';

const StatusOrder = ({ order }) => {
  const dispatch = useDispatch();
  const { orderId } = useParams();

  useEffect(() => {
    dispatch(actionConnectSocket(orderId));
    dispatch(actionGetStatus(orderId));
    dispatch(watchStatusUpdates());
    return () => {
      if (order) {
        dispatch(actionDisconnectSocket(orderId));
      }
    };
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [dispatch, orderId]);

  if (!order) {
    return <div>Loading...</div>;
  }

  return (
    <div>
      <h2>Order Details</h2>
      {console.log(order.status)}
      {/* <p>Order ID: {order.id}</p>
      <p>Status: {order.status}</p> */}
    </div>
  );
};

StatusOrder.propTypes = {
  order: PropTypes.object,
};

const mapStateToProps = createStructuredSelector({
  order: selectOrder,
});

export default connect(mapStateToProps)(StatusOrder);
