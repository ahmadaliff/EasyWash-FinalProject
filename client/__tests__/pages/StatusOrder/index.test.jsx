import StatusOrder from '@pages/StatusOrder';
import { render, fireEvent, act } from '@utils/testHelper';

let wrapper;
const mockNavigate = jest.fn();
const mockDispatch = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
  useParams: () => jest.fn().mockReturnValue({ orderId: 2 }),
}));
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}));
jest.mock('react-to-pdf', () => ({
  usePDF: () => jest.fn(),
}));
jest.mock('@components/MapRouting', () => jest.fn().mockImplementation(() => <div />));
jest.mock('@pages/StatusOrder/selectors', () => ({
  selectOrder: () => ({
    status: 'app_pending',
    id: 2,
    totalPrice: 2000,
    location: '{"lat": -6.224934545453234, "lng": 106.86266439802276 }',
    Services: [
      {
        name: 'reguler',
        isUnit: false,
        servicesOrdered: { price: 2000, quantity: 1 },
        Merchant: { location: '{"lat": -6.224934545453234, "lng": 106.86266439802276 }' },
      },
    ],
  }),
}));
jest.mock('@components/MapLeaflet', () => jest.fn().mockImplementation(() => <div />));

beforeEach(() => {
  act(() => {
    wrapper = render(<StatusOrder />);
  });
});

describe('Order Page', () => {
  test('render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('status-order')).toBeInTheDocument();
  });
  test('navigate back when back button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-back');
    await act(async () => {
      fireEvent.click(button);
    });
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
  test('should be call func handle order', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-cancel');
    await act(async () => {
      fireEvent.click(button);
    });
    expect(mockDispatch).toHaveBeenCalled();
  });
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
