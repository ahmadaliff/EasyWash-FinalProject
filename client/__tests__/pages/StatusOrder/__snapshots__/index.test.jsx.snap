// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
        data-testid="status-order"
      >
        <div
          class="header"
        >
          <h3>
            Detail pesanan
          </h3>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-back"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
          >
            <div
              class="field"
            >
              <p
                class="label"
              >
                Id Pesanan
              </p>
              <p
                class="text"
              >
                2
              </p>
            </div>
            <div
              class="field"
            >
              <p
                class="label"
              >
                Harga Total
              </p>
              <p
                class="text"
              >
                Rp 2.000
              </p>
            </div>
            <div
              class="field"
            >
              <p
                class="label"
              >
                Status
              </p>
              <p
                class="text"
              >
                Ditunda
              </p>
            </div>
            <div
              class="service"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DryCleaningIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19.56 11.36 13 8.44V7c0-.55-.45-1-1-1s-1-.45-1-1 .45-1 1-1 1 .45 1 1h2c0-1.84-1.66-3.3-3.56-2.95-1.18.22-2.15 1.17-2.38 2.35-.3 1.56.6 2.94 1.94 3.42v.63l-6.56 2.92c-.88.38-1.44 1.25-1.44 2.2v.01C3 14.92 4.08 16 5.42 16H7v6h10v-6h1.58c1.34 0 2.42-1.08 2.42-2.42v-.01c0-.95-.56-1.82-1.44-2.21zM18.58 14H17v-1H7v1H5.42c-.23 0-.42-.19-.42-.43 0-.17.1-.32.25-.38l6.75-3 6.75 3c.15.07.25.22.25.39 0 .23-.19.42-.42.42z"
                />
              </svg>
              <div
                class="serviceInfoWrap"
              >
                <hr />
                <div
                  class="serviceInfo"
                >
                  <div
                    class="name"
                  >
                    <p>
                      reguler
                       x 
                    </p>
                    <p
                      class="type"
                    >
                      Tipe  
                      Satuan
                    </p>
                  </div>
                  <p>
                    RpNaN
                  </p>
                </div>
                <hr />
              </div>
            </div>
            <div
              class="field"
            >
              <p
                class="label"
              >
                Lokasi
              </p>
            </div>
            <div />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium cancelButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="button-cancel"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="DoNotDisturbIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8 0-1.85.63-3.55 1.69-4.9L16.9 18.31C15.55 19.37 13.85 20 12 20zm6.31-3.1L7.1 5.69C8.45 4.63 10.15 4 12 4c4.42 0 8 3.58 8 8 0 1.85-.63 3.55-1.69 4.9z"
                  />
                </svg>
              </span>
              Batalkan
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="status-order"
    >
      <div
        class="header"
      >
        <h3>
          Detail pesanan
        </h3>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
          data-testid="button-back"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowBackIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
              />
            </svg>
          </span>
          Kembali
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="field"
          >
            <p
              class="label"
            >
              Id Pesanan
            </p>
            <p
              class="text"
            >
              2
            </p>
          </div>
          <div
            class="field"
          >
            <p
              class="label"
            >
              Harga Total
            </p>
            <p
              class="text"
            >
              Rp 2.000
            </p>
          </div>
          <div
            class="field"
          >
            <p
              class="label"
            >
              Status
            </p>
            <p
              class="text"
            >
              Ditunda
            </p>
          </div>
          <div
            class="service"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="DryCleaningIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19.56 11.36 13 8.44V7c0-.55-.45-1-1-1s-1-.45-1-1 .45-1 1-1 1 .45 1 1h2c0-1.84-1.66-3.3-3.56-2.95-1.18.22-2.15 1.17-2.38 2.35-.3 1.56.6 2.94 1.94 3.42v.63l-6.56 2.92c-.88.38-1.44 1.25-1.44 2.2v.01C3 14.92 4.08 16 5.42 16H7v6h10v-6h1.58c1.34 0 2.42-1.08 2.42-2.42v-.01c0-.95-.56-1.82-1.44-2.21zM18.58 14H17v-1H7v1H5.42c-.23 0-.42-.19-.42-.43 0-.17.1-.32.25-.38l6.75-3 6.75 3c.15.07.25.22.25.39 0 .23-.19.42-.42.42z"
              />
            </svg>
            <div
              class="serviceInfoWrap"
            >
              <hr />
              <div
                class="serviceInfo"
              >
                <div
                  class="name"
                >
                  <p>
                    reguler
                     x 
                  </p>
                  <p
                    class="type"
                  >
                    Tipe  
                    Satuan
                  </p>
                </div>
                <p>
                  RpNaN
                </p>
              </div>
              <hr />
            </div>
          </div>
          <div
            class="field"
          >
            <p
              class="label"
            >
              Lokasi
            </p>
          </div>
          <div />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium cancelButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="button-cancel"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="DoNotDisturbIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8 0-1.85.63-3.55 1.69-4.9L16.9 18.31C15.55 19.37 13.85 20 12 20zm6.31-3.1L7.1 5.69C8.45 4.63 10.15 4 12 4c4.42 0 8 3.58 8 8 0 1.85-.63 3.55-1.69 4.9z"
                />
              </svg>
            </span>
            Batalkan
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
